File: src/ManagementPlatform.Domain/DeadLetterMessage.cs
namespace ManagementPlatform.Domain;

public sealed class DeadLetterMessage
{
    public long Id { get; set; }
    public long OutboxMessageId { get; set; }
    public long CheckoutAttemptId { get; set; }
    public OutboxMessageType Type { get; set; }
    public string PayloadJson { get; set; } = "{}";
    public int AttemptCount { get; set; }
    public string FailureReason { get; set; } = string.Empty;
    public DateTimeOffset FailedAt { get; set; }
}

File: src/main/java/com/managementplatform/domain/model/DeadLetterMessage.java
package com.managementplatform.domain.model;

import com.managementplatform.domain.enums.OutboxMessageType;
import java.time.Instant;

public record DeadLetterMessage(
    long id,
    long outboxMessageId,
    long checkoutAttemptId,
    OutboxMessageType type,
    String payloadJson,
    int attemptCount,
    String failureReason,
    Instant failedAt
) {
}
